tests/models/qwen2/test_tokenization_qwen2.py

```diff
@@ -0,0 +1,134 @@
+# coding=utf-8
+# Copyright 2024 The Qwen Team and the HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import json
+import os
+import unittest
+
+from transformers import Qwen2Tokenizer, Qwen2TokenizerFast
+from transformers.models.qwen2.tokenization_qwen2 import VOCAB_FILES_NAMES, bytes_to_unicode
+from transformers.testing_utils import require_tokenizers
+
+from ...test_tokenization_common import TokenizerTesterMixin
+
+
+@require_tokenizers
+class Qwen2TokenizationTest(TokenizerTesterMixin, unittest.TestCase):
+    tokenizer_class = Qwen2Tokenizer
+    rust_tokenizer_class = Qwen2TokenizerFast
+    test_slow_tokenizer = True
+    test_rust_tokenizer = True
+    space_between_special_tokens = False
+    from_pretrained_kwargs = None
+    test_seq2seq = False
+
```

Comment:
here it's missing some small test to make sure NFC is applied so with special characters + some integration tests!

Reply:
the test cases in (`get_input_output_texts`) include a code point that will fail the test if NFC is not applied (or any other kind of normalization is applied). is it better that I test the prepare_for_tokenization function alone

Reply:
added separate tests for NFC; added integration tests

Reply:
thanks!